resources/hooks/local/pre-push

```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+remote="$1"
+
+while read local_ref local_sha remote_ref remote_sha
+do
+    # Get the branch name
+    local_ref=$(basename $local_ref)
+    break
+done
+    
+# Fetch the GIT diff and format it as command input:
+DIFF=$(git diff --name-only "$remote"/"$local_ref"..HEAD --oneline)
```

Comment:
We need the information about the remote and local branches and this needs to be passed to the tasks as well, so I would suggest to:

- Instead of piping the list of files into the command, pass in the 4 parameters to the `GitPrePushCommand` as options.
- Store the 4 parameters in the `GitPrePushContext` and provide getters for them so that this information is available to tasks: `getLocalSHA1()`, `getLocalRef()`, `getRemoteSHA1()`, `getRemoteSHA1()`.
- Inside `GitPrePushContext::execute()` there are 3 different possibilities for generating the file list:
  1. If the `remote_sha1` is empty (it equals `0000000000000000000000000000000000000000`) then this means we are pushing a brand new branch. We have no comparison point since we don't know which of the ancestor commits have already been checked. In this case we need to check all files.
  2. If the `local_sha1` is empty or if `local_ref` matches the string `(delete)` then this means that the push is intended to delete the remote branch. In this case we should not populate the file list at all, since it is pointless to do any checks on a branch that is about to be deleted.
  3. When both `local_sha1` and `remote_sha1` are not empty, and `local_ref` does not match `(delete)` we are doing a push that updates an already existing remote branch with new code. In this case we can generate a list of files that have changed between the `local_sha1` and `remote_sha1` and can pass it to the context.

To generate the diffs on the PHP side, you can inject the `Repository` dependency into the `PrePushCommand` and then get the file list by calling `$this->repository->run('diff-tree', ['--no-commit-id, ...])`.

It would also be nice to have some helper methods on the context such as `isNewBranch()`, `hasRemoteBranch()`, `isDeletingRemote()`.